Homework-4.py

```diff
@@ -0,0 +1,261 @@
+    pd.set_option("mode.use_inf_as_na", True)
+    df = df.dropna()
+    x = df[predictor_name]
+    y = df[response].map(int)
+
+    predictor = statsmodels.api.add_constant(x)
+    logistic_regression_model = statsmodels.api.Logit(np.asarray(y), np.asarray(predictor))
+    logistic_regression_model_fitted = logistic_regression_model.fit()
+    t_value = round(logistic_regression_model_fitted.tvalues[1], 6)
+    p_value = "{:6e}".format(logistic_regression_model_fitted.pvalues[1])
+    fig = px.scatter(x=x, y=y, trendline="ols")
+    fig.update_layout(
+        title=f"Variable: {predictor_name}: (t-value={t_value}) (p-value={p_value})",
+        xaxis_title=f"Variable: {predictor_name}",
+        yaxis_title="y",
+    )
+    fig.show()
+    stats_table["tvalue"][predictor_name] = t_value
+    stats_table["pvalue"][predictor_name] = p_value
+    print(stats_table)
+
+# step 6. p-value & t-score (continuous predictors only) along with it's plot
+# linear regression will work if response variable it continous
+def linear_regression(df, predictor_name, response, stats_table):
+    x = df[predictor_name]
+    y = df[response]
+
+    predictor = statsmodels.api.add_constant(x)
+    linear_regression_model = statsmodels.api.OLS(y, predictor)
+    linear_regression_model_fitted = linear_regression_model.fit()
+    print(f"Variable: {predictor_name}")
+    print(linear_regression_model_fitted.summary())
+    # Get the stats
+    t_value = round(linear_regression_model_fitted.tvalues[1], 6)
+    p_value = "{:.6e}".format(linear_regression_model_fitted.pvalues[1])
+    # Plot the figure
+    fig = px.scatter(x=x, y=y, trendline="ols")
+
+    fig.update_layout(
+        title=f"Variable: {predictor_name}: (t-value={t_value}) (p-value={p_value})",
+        xaxis_title=f"Variable: {predictor_name}",
+        yaxis_title="y",
+
+    )
+
+    fig.show()
+    stats_table["tvalue"][predictor_name] = t_value
+    stats_table["pvalue"][predictor_name] = p_value
+    print(stats_table)
+
+
+def main():
+    # import data
+
+    global stats_table
+    # step 1. Given titanic pandas dataframe which contains both response and predictor columns
+
+    dataframe = pd.read_csv(
+        "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv"
+    )
+
+    predictors = ["pclass", "sex", "age", "fare", "embarked"]
+    response = "survived"
+
+    dataframe = dataframe[["pclass", "sex", "age", "fare", "embarked", "survived"]]
+
+    # cleaning the dataset
+    dataframe.dropna()
+    dataframe.reset_index(drop=True)
+    dataframe["age"].fillna(dataframe["age"].mean(), inplace=True)
+    dataframe["sex"].replace(["male", "female"], [0, 1], inplace=True)
+    dataframe["embarked"].replace(["S", "C", "Q"], [0, 1, 2], inplace=True)
+    dataframe["embarked"].fillna(dataframe["embarked"].mean(), inplace=True)
+
+    html_table = pd.DataFrame(
+        [0.0, 0.0, 0.0, 0.0, 0.0], index=predictors, columns=["dummy"]
+    )
+
+    stats_table = pd.DataFrame(
```

Comment:
If the dimensions of the csv feature-set increase, the dataframe will give errors.